packages/gatsby-plugin-utils/src/validate.ts

```diff
@@ -0,0 +1,31 @@
+import { BaseValidationOptions, Schema } from "joi"
+
+const validationOptions: BaseValidationOptions = {
+  // Show all errors at once, rather than only the first one every time
+  abortEarly: false,
+  cache: true,
+  // Strip unknown values, this makes sure plugin authors define an extensive schema
+  // TODO: Discuss whether we want this, maybe not a great developer experience...
+  stripUnknown: true,
+}
+
+interface IOptions {
+  validateExternalRules?: boolean
+}
+
+export async function validateOptionsSchema<PluginOptions = object>(
+  pluginSchema: Schema,
+  pluginOptions: PluginOptions,
+  options: IOptions = {}
+): Promise<PluginOptions> {
+  if (options.validateExternalRules === false) {
+    const result = pluginSchema.validate(pluginOptions, {
```

Comment:
Are we able to guarantee that there will never be any async rules, if they're not using `external`?

Reply:
Yes. `external` is explicitly the only way to add async rules.

Reply:
Based on this I'm assuming we default to validating those rules if nothing is explicitly passed?

Reply:
Yes but we could default the other way as well

Reply:
Hard to know what we want at this moment. Seems reasonable to default true.